Sources/NIO/EventLoop.swift

```diff
@@ -291,6 +295,106 @@ extension TimeAmount: Comparable {
     }
 }
 
+extension TimeAmount {
+    public static func + (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds + rhs.nanoseconds)
+    }
+
+    public static func - (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds - rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount.Value, rhs: TimeAmount) -> TimeAmount {
```

Comment:
can we remove all those `*`s? `TimeAmount` (1 ns * 1 ns != 1 ns, it's 1 ns^2)

Reply:
This isn't actually wrong, it's `TimeAmount.Value` * `TimeAmount`, which is to say `UInt64` * `TimeAmount`, which does actually have units of `ns`.

Reply:
Possibly? I think @weissi has stronger opinions than I do on this topic.

Reply:
oh sorry, I totally missed that `lhs` and `rhs` have different types. I think `BinaryInteger` and `TimeAmount` sounds better. Not entirely sure if we really need multiplication here but I'm fine having it too :)